File: Contests/04/D.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        string s;
        cin >> s;
        vector<int> dist;
        for (int i = 0; i < n - 1; i++) {
            bool l = false;
            for (int j : dist) {
                if (s.substr(0, i) + s.substr(i + 2) == s.substr(0, j) + s.substr(j + 2)) {
                    l = true;
                    break;
                }
            }
            if (!l) {
                dist.push_back(i);
            }
        }
        cout << dist.size() << endl;
    }
    return 0;
}

File: Contests/04/D.py
t = int(input())

for _ in range(t):
    n = int(input())
    s = input()
    dist = list()
    for i in range(n-1):
        l = 0
        for j in dist:
            if s[:i] + s[i+2:] == s[:j] + s[j+2:]:
                l = 1
                break
        if not l:
            dist.append(i)
    print(len(dist))
